test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -55,7 +55,15 @@ async def test_table_desc_read_barrier(request, manager: ManagerClient) -> None:
     async with inject_error(manager.api, servers[0].ip_addr, 'group0_state_machine::delay_apply'):
         logger.info("Altering table")
         await cql.run_async("alter table ks.t add s1 int", host=hosts[1])
-        # TODO: trigger the read barrier once available
-        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
 
+        # verify that there is a schema difference
+        # (raft apply delayed, so the first server doesn't have the new schema yet)
+        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
         assert desc_schema[0] != desc_schema[1]
+
+        # waits for the schema fully updated on all nodes (after the delay ends)
+        await read_barrier(manager.api, servers[0].ip_addr)
+
+        # verify that there is no schema difference after the read barrier
+        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
```

Comment:
I have to admit I got confused here. Since "DESC SCHEMA" is an official CQL operation that queries, well,..., the schema, isn't its responsibility to wait until the node is sure about the schema? Why doesn't DESC SCHEMA do its own "read barrier" if that is needed to make read of the schema?

Reply:
As I understand according to the explanation in #19213 it only does "schema agreement wait", which is only done locally and is what causes the problem with old schema version, as explained here:

> Problem with awaiting schema agreement is that it is usually done from a single node POV. Gocql implementation does that by checking if schema versions from system.local/peers are the same, but it does so only from the control connection POV

Reply:
IIUC `DESC SCHEMA` does not do any wait. The aforementioned wait is something that driver does after performing a schema **change**, but as the issue explains, it is not enough for the backup use case.

Why doesn't `DESC SCHEMA` do its own barrier? Well, it never did, it was always returning local schema. And I think that's fine; we might sometimes want to check local schema of a node without forcing it to communicate with quorum. That would be a backwards-incompatible interface change and potentially eliminate some valid use cases. If you want to ensure you obtain recent schema, you can call read_barrier explicitly. I think that's fine.